src/constraint/spec.ts

```diff
@@ -543,8 +549,10 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
       for (let encQ of encodings) {
         if (encQ.scale) {
           const scale: ScaleQuery = encQ.scale as ScaleQuery;
-          if (contains([ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC], scale.type) &&
-             (scale.zero === true)) {
+
+          if (contains([ScaleType.LOG, ScaleType.ORDINAL, ScaleType.TIME, ScaleType.UTC],
+              scaleType(scale.type as ScaleType, encQ.timeUnit as TimeUnit, encQ.type as Type)) &&
```

Comment:
assign `const sType = ...`   This line is getting too long.  (But no need for declaration for each parameter.)